src/test/regress/expected/with_modifying.out

```diff
@@ -596,11 +596,7 @@ raw_data AS (
 	DELETE FROM modify_table WHERE id = 1 AND val IN (SELECT val FROM select_data) RETURNING *
 )
 SELECT COUNT(*) FROM raw_data;
- count 
--------
-     2
-(1 row)
-
+ERROR:  set-valued function called in context that cannot accept a set
```

Comment:
We're hitting an already existing issue by query transformation: https://github.com/citusdata/citus/issues/3189#issuecomment-571497051

I'll fix the query as suggested above for now

Reply:
This doesn't really fix the underlying issue right? This PR still has the regression that we stop supporting this pattern in CTEs, so people their queries might stop working if we merge this and they update. 
A workaround for those people would be to disable cte inlining for such a query with `citus.enable_cte_inlining`. 

If we want to go this way, then that should be a concious decision. If not, we should not merge this until we fix #3189 or find a workaround in this PR so we don't hit that bug (e.g. not doing CTE inlining for CTEs with `now` or other functions like that in there)

Changing the query in the test isn't really either of those. It simply means we're ignoring the issue.

Reply:
> This doesn't really fix the underlying issue right?

No, it doesn't  fix  the issue.

> If we want to go this way, then that should be a concious decision. If not, we should not merge this until we fix #3189 or find a workaround in this PR so we don't hit that bug (e.g. not doing CTE inlining for CTEs with now or other functions like that in there)

To me, the  issue we  hit is an edge case, so I wouldn't mind merging the PR and prioritize fixing the other later on. 

As far as I can  tell, the benefits of this PR is large enough  to ignore this edge case, where there are multiple workarounds to suggest, in case anyone hit by this.

Reply:
>Changing the query in the test isn't really either of those. It simply means we're ignoring the issue.

I just didn't want to change an  existing  test written for some other purpose

Reply:
> As far as I can tell, the benefits of this PR is large enough to ignore this edge case, where there are multiple workarounds to suggest, in case anyone hit by this.

Alright that sounds reasonable. @marcocitus do you agree with that?

Reply:
> Alright that sounds reasonable. @marcocitus do you agree with that?

I'm also curious about what Marco would think about this.

Reply:
Like we discussed in person with marco, this seems acceptable. Let's confirm that the workaround works though, by adding the old test back, but with `SET citus.enable_cte_inlining TO FALSE`.